projects/scannet_offline_eval/demo/proto_robot/robodata_server.py

```diff
@@ -36,14 +43,46 @@ def GetHistory(self, request, context):
         #     x = robodata_pb2.RobotSummary()
         #     x.message = "hello"
         #     return x
-        breakpoint()
-        return self.obs_history.pop(0)
+        yield self.obs_history.pop(0)
+
+    def PlanHighLevelAction(self, request, context):
+        req_msg = ""
+        obj_crops = []
+        for r in request:
+            req_msg = req_msg + r.message
+            for im in r.instance_image:
+                tensor_img = self._robotensor_to_tensor(im)
+                print(tensor_img.shape)
+                post_img = self.predictor.vis_processor(
+                    Image.fromarray(np.array(tensor_img).astype(np.uint8))
+                )
+                print(post_img.shape)
+                obj_crops.append(post_img)
+        prompt = "Task: You are a chatbot exploring a home."
+        # TODO something more sophisticated
+        prompt = prompt + req_msg
+        chat_input = {
+            "prompt": [prompt],
+            "crops": torch.stack(obj_crops, dim=0).unsqueeze(0),
+            "images": [],
+        }
+        print("len of crops:" + str(len(obj_crops)))
```

Comment:
Logger pls for anything server-related. Otherwise it's so hard to figure out where a print is coming from

Reply:
We should start doing this more generally anyway